src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -117,28 +125,51 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     }
   }
 
+  private def reportLagTime(dataFrame: DataFrame, reportLagTimeColumn: Option[String], instrumentationProvider: InstrumentationProvider) ={
+    reportLagTimeColumn match {
+      case Some(timeColumn) => {
+        val currentTime = System.currentTimeMillis
+        val timeColumnType = dataFrame.schema.filter(f => f.name == timeColumn).map(f => f.dataType).last
+        try {
+        val maxDataframeTime = timeColumnType match {
+          case timeColumnType:TimestampType => dataFrame.agg({timeColumn.toString -> "max"}).collect()(0).getTimestamp(0).getTime()
+          case _ =>   dataFrame.agg({timeColumn.toString -> "max"}).collect()(0).getLong(0)
+        }
+          instrumentationProvider.gauge(name = "lag", currentTime - maxDataframeTime)
+        } catch {
+          case e: ClassCastException => {
+            throw new ClassCastException(s"Lag instrumentation column -${timeColumn} cannot be cast to spark.sql.Timestamp or spark.sql.Long")
+          }
+        }
+      }
+
+      }
+    }
+
   def write(job: Job): Unit = {
+
     configuration.output match {
       case Some(output) => {
-        val streamingWriterList: mutable.Map[String, StreamingWritingConfiguration] = mutable.Map()
-
+        val streamingWriterList: mutable.Map[String, (StreamingWritingConfiguration, Output)] = mutable.Map()
         output.foreach(outputConfig => {
           val writer = WriterFactory.get(outputConfig, metricName, job.config, job)
           val dataFrameName = outputConfig.dataFrameName
-          val dataFrame = repartition(outputConfig, job.sparkSession.table(dataFrameName))
+          var dataFrame = repartition(outputConfig, job.sparkSession.table(dataFrameName))
 
           if (dataFrame.isStreaming) {
-            val streamingWriterConfig = streamingWriterList.getOrElse(dataFrameName, StreamingWritingConfiguration(dataFrame))
-            streamingWriterConfig.writers += writer
+            val streamingWriterConfig = streamingWriterList.getOrElse(dataFrameName, (StreamingWritingConfiguration(dataFrame), outputConfig))
+            streamingWriterConfig._1.writers += writer
```

Comment:
Let's do something else please.. :)
Maybe a new case class with all the things you need for streaming